### PR TITLE
Update Architecture.md

### PR DESCRIPTION
**Pull request by nirinchev**

### CODE REVIEW

**Comment by nirinchev** [comment]
@papafe I believe I've addressed all your comments. To better reflect their roles, I renamed `RemoteFileManager` to `RemoteStorageManager` and `FileManager` to `LFSManager` as that was more of a "everything manager" rather than just concerned with local files. LMK if that makes sense to you. I renamed the `FunctionsFileManager` to `AtlasFunctionsStorageManager` but didn't rename the package as I didn't want to deal with package deprecations and stuff.

File: Architecture.md [hunk 1]
```diff
@@ -1,32 +1,46 @@
...
-4. We update the `FileData` with the url of the newly uploaded file ([code link](https://github.com/realm/realm-dotnet-lfs/blob/630b323b31e45a25fee9fac4bf745c8b9123c34a/Realm.LFS/Managers/RemoteFileManager.cs#L143-L156))
+![Architecture diagram](https://github.com/realm/realm-dotnet-lfs/assets/2315687/fb0ade9d-a0f0-4f73-952e-ad7917e16a9d)
+
+The diagram above roughly describes the functionality of the library, but the general flow is as follows:
+
+1. User creates `FileData` object ([code link](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/FileData.cs#L70))
+2. We store the Stream to a file on disk ([code link](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/FileData.cs#L72))
+3. We try to upload the file (with incremental backoffs on failure) ([code link](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/FileData.cs#L83))
+4. We update the `FileData` with the url of the newly uploaded file ([code link](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/Managers/RemoteStorageManager.cs#L143-L156))
```
Comment (papafe) [inline Architecture.md:10]: This link is not working, but I suppose it should once it's all merged to main. Probably we should give a final "click review" for the links once it's merged 😁

File: README.md [hunk 2]
```diff
@@ -11,9 +11,9 @@ Using binary data (i.e. `byte[]` properties) with Realm is supported but very in
 To initialize the SDK, the minimum configuration you need to do is to configure the remote manager factory:
 
 ```csharp
-FileManager.Initialize(new FileManagerOptions
+LFSManager.Initialize(new LFSOptions
 {
-    RemoteManagerFactory = (config) => new FunctionsFileManager(config, "MyDataFunction")
+    RemoteManagerFactory = (config) => new FunctionsStorageManager(config, "MyDataFunction")
```
Comment (papafe) [inline README.md:16]:
```suggestion
    RemoteManagerFactory = (config) => new AtlasFunctionsStorageManager(config, "MyDataFunction")
```

File: Realm.LFS.Functions/README.md [hunk 3]
```diff
@@ -28,9 +28,9 @@ public class Recipe : RealmObject
 To initialize the SDK, the minimum configuration you need to do is to configure the remote manager factory:
 
 ```csharp
-FileManager.Initialize(new FileManagerOptions
+LFSManager.Initialize(new LFSOptions
 {
-    RemoteManagerFactory = (config) => new FunctionsFileManager(config, "MyDataFunction")
+    RemoteManagerFactory = (config) => new FunctionsStorageManager(config, "MyDataFunction")
```
Comment (papafe) [inline Realm.LFS.Functions/README.md:33]:
```suggestion
    RemoteManagerFactory = (config) => new AtlasFunctionsStorageManager(config, "MyDataFunction")
```

**Review by papafe — APPROVED** [review]
Looks good! 
Maybe it would be worth suggesting how users should "wait for the image being uploaded on another device" part.

File: Architecture.md [hunk 4]
```diff
@@ -1,32 +1,46 @@
...
+```
+
+This is the RealmObject that holds the metadata associated with the file. It is constructed with a binary source, which it then [persists in a temporary location](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/FileData.cs#L59). When the object is added to Realm, we [schedule the upload](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/FileData.cs#L70).
 
-## [`FileManager`](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/Managers/FileManager.cs)
+## [`LFSManager`](https://github.com/realm/realm-dotnet-lfs/blob/main/Realm.LFS/Managers/LFSManager.cs)
 
 This is the class that contains most of the orchestration logic and for the most part deals with local files. It exposes internal methods for filesystem manipulation (reading, deleting, etc.), as well as enqueuing files for upload. There are two main file locations:
 
 1. `Temporary`: this is where we store all files for `FileData`-s that were created, but have not yet been added to Realm.
```
[the remark this answers is not in the document]
Comment (nirinchev) [reply Architecture.md:33]: Yep, will clarify

**Pull request merged**